admin/configuration.php

```diff
@@ -58,13 +58,14 @@
     $cfg_group->fields['configuration_group_title'] = '';
 } else {
     // multilanguage support:
-    // For example, in admin/includes/languages/spanish/configuration.php
+    // For example, in admin/includes/languages/spanish/lang.configuration.php
     // define('CFG_GRP_TITLE_MY_STORE', 'Mi Tienda');
     $str = $cfg_group->fields['configuration_group_title'];
-    $str = preg_replace('/[^a-zA-Z0-9_\x80-\xff]/', '_', $str);
-    $const = 'CFG_GRP_TITLE_' . strtoupper($str);
-    if (defined($const)) {
-        $cfg_group->fields['configuration_group_title'] = constant($const);
+    $str = str_replace(' ', '_', $str);
+    $str = strtoupper(preg_replace('/[^a-zA-Z0-9_\x80-\xff]/', '', $str));
```

Comment:
Why did you replace the `_` substitution with a blank?

Reply:
In POSM, some text/title use quote ' which was replaced by _ making constant NAMES__LIKE_THIS (2 _) which does not work well.

Reply:
Hmmm. Good point. Could also run another preg_replace afterward, converting `_+` (means 1 or more in a row) to `_` (single).

Reply:
Yes, I thought of it, but the logic of changing any 'special' character into _ was kind of reversed. Only space that separate words should be changed and other special characters removed.
Still, in case some title contains two or more space in a row, it might be safer to add this line:
$str = preg_replace('/_+/', '_', $str);

Or perhaps better replace the first line (str_replace) by this:
$str = preg_replace('/[\s ]+/', '_', $str);